### PR TITLE
Fixes #7009 - Adds info about PSAnsiRendering.

### LINKED ISSUE #7009: ExperimentalFeature: PSAnsiRendering
# Summary of the new document or enhancement

- List of articles that need to be updated: 
   - https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/docs-conceptual/learn/experimental-features.md
   - https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md
- New articles that need to be created:
- Link(s) to related code PR(s) in the PowerShell/PowerShell repo: https://github.com/PowerShell/PowerShell/pull/13758
- Link(s) to related issue(s) in the PowerShell/PowerShell repo: https://github.com/PowerShell/PowerShell/issues/13071

### PR DESCRIPTION
**Pull request by chasewilson**
# PR Summary

Adds information about new experimental feature - PSAnsiRendering

## PR Context

Fixes #7009 
Fixes [AB#1801532](https://dev.azure.com/mseng/677da0fb-b067-4f77-b89b-f32c12bb8617/_workitems/edit/1801532)

Select the area of the Table of Contents containing the documents being changed.

**Conceptual content**
- [x] Overview and Install
- [ ] Learning PowerShell
  - [ ] PowerShell 101
  - [ ] Deep dives
  - [ ] Remoting
- [ ] Release notes (What's New)
- [ ] Windows PowerShell
  - WMF, ISE, release notes, etc.
- [ ] DSC articles
- [ ] Community resources
- [ ] Sample scripts
- [ ] Gallery articles
- [ ] Scripting and development
  - [ ] Legacy SDK

**Cmdlet reference & about_ topics**
- [x] Preview content
- [ ] Version 7.1 content
- [ ] Version 7.0 content
- [ ] Version 5.1 content

## PR Checklist

- [x] I have read the [contributors guide][contrib] and followed the style and process guidelines
- [x] PR has a meaningful title
- [x] PR is targeted at the _staging_ branch
- [x] All relevant versions updated
- [x] Includes content related to issues and PRs - see [Closing issues using keywords][key].
- [x] This PR is ready to merge and is not **Work in Progress**
  - If the PR is work in progress, please add the prefix `WIP:` or `[WIP]` to the beginning of the
    title and remove the prefix when the PR is ready.

[contrib]: https://docs.microsoft.com/powershell/scripting/community/contributing/overview
[key]: https://help.github.com/en/articles/closing-issues-using-keywords

### CODE REVIEW

**Comment by chasewilson** [comment]
@sdwheeler  I didn't have a description for the `AttributesOff` property in the table. And If I'm missing somnething feel free to let me know

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md [hunk 1]
```diff
@@ -492,6 +492,69 @@ from the originating session. To add data to the **ApplicationArguments**
...
+> This is an experimental feature. For more information see
+> [about_Experimental_Features](about_Experimental_Features.md).
+
+As of PowerShell 7.2 you can now access the `$PSStyle` automatic variable to
+view and change the rendering of ANSI string output. The variable contains the
+following properties:
+
+| Property        | Description                                                |
+| --------------- | ---------------------------------------------------------- |
+| AttributesOff   | |
```
Comment (SteveL-MSFT) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md:507]: This was renamed to `Reset` which turns off all decoration (colors, styles, etc...)

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md [hunk 2]
```diff
@@ -492,6 +492,69 @@ from the originating session. To add data to the **ApplicationArguments**
...
+|                 | `Host.PrivateData` such as errors, warnings, etc. In       |
+|                 | addition to colring, you can also control extra attributes |
+|                 | like bolding and underlining as well.                      |
+| Hidden          | Turns Hidden on.                                           |
+| HiddenOff       | Turns Hidden off.                                          |
+| OutputRendering | This is a nested `OutputRendering` object to control output|
+|                 | rendering.                                                 |
+| Reverse         | Turns Reverse on.                                          |
+| ReverseOff      | Turns Reverse off.                                         |
+| Standout        | Turns Standout on.                                         |
```
Comment (SteveL-MSFT) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md:524]: This was renamed to Italic, same for ItalicOff

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md [hunk 3]
```diff
@@ -492,6 +492,69 @@ from the originating session. To add data to the **ApplicationArguments**
...
+| StandoutOff     | Turns Standout off.                                        |
+| Underlined      | Turns Underlined on.                                       |
```
Comment (SteveL-MSFT) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md:526]: This was renamed to `Underline` (no trailing `d`).  Same for off.

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md [hunk 4]
```diff
@@ -492,6 +492,69 @@ from the originating session. To add data to the **ApplicationArguments**
...
+| UnderlinedOff   | Turns Underlined off.                                      |
+
+The base members return ANSI escape sequences mapped to their names. These are
+also settable so you can change bold to underlined, for example. This makes it
+easier for you to author decorated strings with tab completion:
+
+```powershell
+"$($PSStyle.Background.LightCyan)Power$($PSStyle.Underlined)$($PSStyle.Bold)Shell$($PSStyle.AttributesOff)"
```
Comment (SteveL-MSFT) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md:534]: Update to use `$PSStyle.Reset`

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md [hunk 5]
```diff
@@ -492,6 +492,69 @@ from the originating session. To add data to the **ApplicationArguments**
...
+```
+
+The `$PSStyle.Background` and `$PSStyle.Foreground` members contain the
+standard 16 consol colors as well as `RGB()` methods to specify 24-bit color.
```
Comment (SteveL-MSFT) [inline reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md:538]:
```suggestion
standard 16 console colors as well as `RGB()` methods to specify 24-bit color.
```

File: reference/docs-conceptual/learn/experimental-features.md [hunk 6]
```diff
@@ -73,6 +74,45 @@ In this example, a job is started and a breakpoint is set to break when the `Set
...
+
+In PowerShell 7.2, this experiment enables the `$PSStyle` automatic variable to control ANSI
+rendering of string output.
+
+```powershell
+PS> $PSStyle
+
+Name            MemberType Definition
+----            ---------- ----------
+AttributesOff   Property   string AttributesOff {get;set;}
```
Comment (SteveL-MSFT) [inline reference/docs-conceptual/learn/experimental-features.md:87]: Same edits as above

File: reference/docs-conceptual/learn/experimental-features.md [hunk 7]
```diff
@@ -73,6 +74,45 @@ In this example, a job is started and a breakpoint is set to break when the `Set
...
+
+The base members return ANSI escape sequences mapped to their names. These are also settable so you
+can change bold to underlined, for example. This makes it easier for scripters to author decorated
+strings with tab completion:
+
+```powershell
+"$($PSStyle.Background.LightCyan)Power$($PSStyle.Underlined)$($PSStyle.Bold)Shell$($PSStyle.AttributesOff)"
+```
+
+For more information, see [about_automatic_variables](/reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md)
```
Comment (SteveL-MSFT) [inline reference/docs-conceptual/learn/experimental-features.md:114]: It's probably worth mentioning that for C# developers, they need to access `PSStyle` which is a singleton, so usage would be like:

```c#
string output = $"{PSStyle.Instance.Foreground.Red}{PSStyle.Instance.Bold}Hello{PSStyle.Instance.Reset}";
```

`PSStyle` exists in the System.Management.Automation namespace.

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

File: reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md (hunk 1 above)
Comment (chasewilson) [reply reference/7.2/Microsoft.PowerShell.Core/About/about_Automatic_Variables.md:507]: Much appreciated! Fixed

File: reference/docs-conceptual/learn/experimental-features.md (hunk 7 above)
Comment (chasewilson) [reply reference/docs-conceptual/learn/experimental-features.md:114]: Added

**Review by chasewilson — COMMENTED** [review]
@sdwheeler do you have any changes you would like to see? @SteveL-MSFT All cahnges should have been made. If you see anything else you want to add, let me know.

**Review by sdwheeler — CHANGES_REQUESTED** [review]
I made changes but there is information missing.

Please add the information about the engine changes to the experimental-features.md article.

**Review by sdwheeler — APPROVED** [review]

**Pull request merged**